File: p2p_library/src/main/java/com/tambapps/p2p/fandem/listener/TransferListener.java
package com.tambapps.p2p.fandem.listener;

import com.tambapps.p2p.fandem.Peer;

/**
 * A listener of sharing events
 */
public interface TransferListener {

  void onConnected(Peer selfPeer, Peer remotePeer, String fileName, long fileSize);

  void onProgressUpdate(int progress, long bytesProcessed, long totalBytes);

}

File: p2p_library/src/main/kotlin/com/tambapps/p2p/fandem/listener/TransferListener.kt
package com.tambapps.p2p.fandem.listener

import com.tambapps.p2p.fandem.Peer

/**
 * A listener of sharing events
 */
interface TransferListener {

    fun onConnected(selfPeer: Peer, remotePeer: Peer, fileName: String, fileSize: Long)

    fun onProgressUpdate(progress: Int, bytesProcessed: Long, totalBytes: Long)

}